From c7911e4032bc431c3aa67af6c91a9b25b4c81a46 Mon Sep 17 00:00:00 2001
From: auouymous <au@qzx.com>
Date: Mon, 12 Sep 2022 14:16:31 -0600
Subject: [PATCH] Fix crash on older versions of python.

Older versions of python do not treat None as False. This causes
set_sensitive() to crash when None is returned from can_pause().

Fixes #1394.
---
 src/gpodder/model.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/gpodder/model.py b/src/gpodder/model.py
index 0e8f9782..970f8e56 100644
--- a/src/gpodder/model.py
+++ b/src/gpodder/model.py
@@ -497,7 +497,7 @@ class PodcastEpisode(PodcastModelObject):
         PAUSING and PAUSED tasks can be resumed.
         """
         return not self.was_downloaded(and_exists=True) and (
-            not self.download_task
+            self.download_task is None
             or self.download_task.can_queue()
             or self.download_task.status == self.download_task.PAUSING)
 
@@ -505,20 +505,20 @@ class PodcastEpisode(PodcastModelObject):
         """
         gPodder.on_pause_selected_episodes() filters selection with this method.
         """
-        return self.download_task and self.download_task.can_pause()
+        return self.download_task is not None and self.download_task.can_pause()
 
     def can_cancel(self):
         """
         DownloadTask.cancel() only cancels the following tasks.
         """
-        return self.download_task and self.download_task.can_cancel()
+        return self.download_task is not None and self.download_task.can_cancel()
 
     def can_delete(self):
         """
         gPodder.delete_episode_list() filters out locked episodes, and cancels all unlocked tasks in selection.
         """
         return self.state != gpodder.STATE_DELETED and not self.archive and (
-            not self.download_task or self.download_task.status == self.download_task.FAILED)
+            self.download_task is None or self.download_task.status == self.download_task.FAILED)
 
     def can_lock(self):
         """
-- 
2.37.2

